src/Native/Unix/System.Native/pal_networking.cpp

```diff
@@ -1144,13 +1144,25 @@ SystemNative_SetIPv6Address(uint8_t* socketAddress, int32_t socketAddressLen, ui
     return PAL_SUCCESS;
 }
 
-static void ConvertMessageHeaderToMsghdr(msghdr* header, const MessageHeader& messageHeader)
+static bool IsStreamSocket(int socket)
+{
+    int type;
+    socklen_t length = sizeof(int);
+    return socket != -1
```

Comment:
Is this socket != -1 check needed?  I'd have assumed that if an erroneous value snuck in, getsockopt would just return EBADF or something like that.

Reply:
I'll remove this.